tests/src/performance/perflab/LowLevelPerf.cs

```diff
@@ -0,0 +1,765 @@
+        string aString = "foo";
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                aGenericClassWithString.aGenericInstanceFieldT = aString;
+    }
+
+    [Benchmark]
+    public static int ForeachOverList100Elements()
+    {
+        List<int> iList = new List<int>();
+        for (int i = 0; i < 100; i++)
+            iList.Add(i);
+
+        int iResult = 0;
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                foreach (int i in iList)
+                    iResult = i;
+
+        return iResult;
+    }
+
+    [Benchmark]
+    public static Type TypeReflectionObjectGetType()
+    {
+        Type type = null;
+        object anObject = "aString";
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                type = anObject.GetType();
+
+        return type;
+    }
+
+    [Benchmark]
+    public static Type TypeReflectionArrayGetType()
+    {
+        Type type = null;
+        object anArray = new string[0];
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                type = anArray.GetType();
+
+        return type;
+    }
+
+    [Benchmark]
+    public static void ReaderWriterLockSlimPerf()
+    {
+        ReaderWriterLockSlim rwLock = new ReaderWriterLockSlim();
+
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            using (iteration.StartMeasurement())
+            {
+                rwLock.EnterReadLock();
+                rwLock.ExitReadLock();
+            }
+        }
+    }
+
+    [Benchmark]
+    public static Guid SlowGuidCreation()
+    {
+        Guid g = default(Guid);
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                g = new Guid("044973cd-251f-4dff-a3e9-9d6307286b05");
+
+        return g;
+    }
+
+    [Benchmark]
+    public static DateTime DateTimeNow()
+    {
+        DateTime now = default(DateTime);
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                now = DateTime.Now;
+
+        return now;
+    }
+
+    [Benchmark]
+    public static DateTime DateTimeUtcNow()
+    {
+        DateTime now = default(DateTime);
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                now = DateTime.UtcNow;
+
+        return now;
+    }
+
+    [Benchmark]
+    public static string IntegerFormatting()
+    {
+        int number = Int32.MaxValue;
+
+        string result = null;
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                result = number.ToString();
+
+        return result;
+    }
```

Comment:
This one should stay here as well, as the implementation is mostly in the runtime.  I'm OK leaving it in both places to ensure we cover the surface in corefx as well, though there's not too much there.